### PR TITLE
 improve code readability

### PR DESCRIPTION
**Pull request by wangmiao1981**
Add a few items in gitignore;

For BaseSnapshot::cacheManifests, it code implies that deleteManifests will always be set. This change is just to explicitly set deleteManifests if it is null to increase code readability.

Tests: Existing UTs in Core. No need for new UTs.

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/BaseSnapshot.java [hunk 1]
```diff
@@ -130,6 +130,9 @@ private void cacheManifests() {
     if (dataManifests == null) {
       this.dataManifests = ImmutableList.copyOf(Iterables.filter(allManifests,
           manifest -> manifest.content() == ManifestContent.DATA));
+    }
+
+    if (deleteManifests == null) {
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/BaseSnapshot.java:135]: What about using `dataManifests == null || deleteManifests == null`?

**Comment by rdblue** [comment]
@wangmiao1981, could you separate these into different PRs? We try to keep changes focused on just one problem so that they can be cherry-picked between branches without unnecessary git conflicts.

**Comment by wangmiao1981** [comment]
I will make the change to 2 PRs. Thanks!

**Comment by rdblue** [comment]
Merged. Thanks @wangmiao1981!

**Pull request merged**